winch/codegen/src/isa/x64/masm.rs

```diff
@@ -2543,6 +2543,183 @@ impl Masm for MacroAssembler {
         Ok(())
     }
 
+    fn v128_trunc_sat(
+        &mut self,
+        context: &mut CodeGenContext<Emission>,
```

Comment:
Usually `CodeGenContext` is passed when there's a special ISA needs for lowering, but I don't think that the case here? If I'm reading the code correctly, it seems that the only reason why we need the context is to pop a temporary register when kind == `F32x4U`?   

If that's the case, could we instead avoid passing the context and pass an `Option<Reg>` at each call site where appropriate?

Reply:
Would we still want to allocate the extra register on AArch64 for `F32x4U` when it isn't necessary on that ISA? 

Otherwise I'm not sure how to handle only allocating the register on x64 if we're doing the allocation outside the x64 macroassembler.

Reply:
> Would we still want to allocate the extra register on AArch64 for F32x4U when it isn't necessary on that ISA?

I was under the impression that we'd still need an extra reg for aarch64, is that not the case? More generally, maybe it's fine to leave it as is and we can refactor this once we add SIMD support for other backends.

Reply:
AFAICT, we could implement this instruction with just a single emission of [FCVTZU](https://developer.arm.com/documentation/dui0801/l/A64-Floating-point-Instructions/FCVTZU--scalar--fixed-point---A64-) on AArch64 and we would only need the one register.